data/blog/tailgraft.mdx

```diff
@@ -0,0 +1,179 @@
+
+```
+ssh ubuntu@mincemeat
+```
+
+No passwords or keys required! You can do whatever you like from here
+from setting up [Tailscale
+Serve](https://tailscale.com/kb/1242/tailscale-serve/) to point to a
+local [Gitea](https://gitea.io) install, setting up an internal wiki,
+or anything you can imagine. You can even expose things to the public
+internet with [Funnel](https://tailscale.dev/blog/funnel-101).
+
+<ConvSnippet authors={props.authorDetails} name="Xe Iaso">
+  If you chose to make your Pi an exit node, you will need to [confirm that in the admin
+  console](https://tailscale.com/kb/1103/exit-nodes/#step-2-allow-the-exit-node-from-the-admin-console).
+  Once you've done that, you can enable the exit node in the Tailscale menulet, system tray icon, or
+  in the app.
+</ConvSnippet>
+
+## How it works
+
+Now that you have your Pi set up, let's take a moment to learn how
+this all works and why I said it's using the magic of the cloud.
+
+When you set up an instance in AWS or a droplet in Digital Ocean, you
+don't usually just get dumped a username and password and told "good
+luck". There's ways to pre-load SSH keys and other configuration via
+the "user data" field. This allows you to use everyone's favorite
+configuration language to tell the platform how it should mangle the
+base image to your needs: YAML. This is powered by
+[cloud-init](https://cloud-init.io/), a tragically under-documented
+critical part of the modern internet.
+
+Now you may be wondering something like:
+
+<XesiteConv name="Aoi" mood="wut">
+  Okay, wait. You just said that we're going to be using Ubuntu on a Raspberry Pi in order to do
+  this. How is the cloud relevant?
+</XesiteConv>
+
+It's relevant beacuse the Ubuntu Raspberry Pi images ships with
+cloud-init! This lets you steal the YAML fire from the cloud gods so
+that you can assimilate your pi into your tailnet. Not to mention,
+it's actually possible to buy Raspberry Pi boards without having to
+sign over your kidneys or hunting them out from scalpers like we did
+with PS5s.
+
+> And when Prometheus stole the YAML from the gods, they cried out
+> "NO!". But Prometheus did not understand, for all he heard was
+> "false".
+
+### cloud-init installs Tailscale
+
+One of the things cloud-init offers is the `runcmd` block. This allows
+you to make a list of arbitrary shell commands that are run as root
+some time after boot. Here's an example `runcmd` block that the shell
+script makes:
+
+```yaml
+runcmd:
+  - ['sh', '-c', 'curl -fsSL https://tailscale.com/install.sh | sh']
+  - [
+      'sh',
+      '-c',
+      "echo 'net.ipv4.ip_forward = 1' | sudo tee -a /etc/sysctl.d/99-tailscale.conf && echo 'net.ipv6.conf.all.forwarding = 1' | sudo tee -a /etc/sysctl.d/99-tailscale.conf && sudo sysctl -p /etc/sysctl.d/99-tailscale.conf",
+    ]
+  - ['tailscale', 'up', '--ssh', '--advertise-exit-node', '--authkey=changeme', '--hostname=ubuntu']
+  - ['sh', '-c', 'sudo hostnamectl hostname ubuntu']
+```
+
```

Comment:
And then here we might say something like

If you're on Windows or couldn't get the script to add this automatically, you can add the text above to the tktktk file manually. Be sure to add in your own auth key.

Or similar